### ISSUE TITLE
TypeError: Cannot read property 'includes' of null

### ISSUE THREAD

**Issue by Lewiscowles1986** [adamisntdead/poke#1]
I ran this on a public website. It took quite some time to crash with this failure.

```
(node:26132) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'includes' of null
(node:26132) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

No clue why this has happened, as on another website it completed a lot faster.

**Comment by adamisntdead** [comment]
Okay, I will look into that, thanks!


On March 17, 2018 at 1:41:25 PM, Lewis Cowles (notifications@github.com) wrote:

I ran this on a public website. It took quite some time to crash with this failure.

(node:26132) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'includes' of null
(node:26132) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
No clue why this has happened, as on another website it completed a lot faster.

**Comment by Lewiscowles1986** [comment]
Right I've patched and at least I get it to complete running (106minutes lol)

```
{ Error: Parse Error
    at TLSSocket.socketOnData (_http_client.js:440:20)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at TLSSocket.Readable.push (_stream_readable.js:208:10)
    at TLSWrap.onread (net.js:594:20) bytesParsed: 0, code: 'HPE_INVALID_CONSTANT' }
```

Comes up in output three times, I've asked and I'm not allowed to share the URL's (sorry)

The only link I'm able to share is the following

```
Missing / Broken Content Found :(
 - https://manage.kmail-lists.com/subscriptions/subscribe
```

The basic problem is addressed in the PR linked above. `url.parse`  dependency seems to generate `null` for some URL's. It's likely an indicator of a problem, maybe should be caught and return a message for logs, but until I make that modification, #2 is allowing me to complete scanning the system.

**Comment by adamisntdead** [comment]
Okay, I merged that, I'm going to take a look at that now, thanks for the contribution

**Comment by adamisntdead** [comment]
I looked through, and I found that there is a url object module thats causing some problems. I updated all of the modules, and everything seems to work on my end. I will set up some CI testing too, but this will be fixed in the next commit. Thanks so much for everything!

**Comment by adamisntdead** [comment]
Solved in version 1.1.0

**Comment by Lewiscowles1986** [comment]
Rockstar!
